metrics-proxy/src/test/java/ai/vespa/metricsproxy/service/MetricsParserTest.java

```diff
@@ -24,6 +25,21 @@ public void consume(Metric metric) {
         }
     }
 
+    @Test
+    public void dimensions_hashcode_is_different_for_distinct_but_duplicate_dimension_values() {
+        var dimensions1 = List.of(
+                new MetricsParser.Dimension("cluster", "CLUSTER-1"),
+                new MetricsParser.Dimension("clusterid", "CLUSTER-1"));
+
+        var dimensions2 = List.of(
+                new MetricsParser.Dimension("cluster", "CLUSTER-2"),
+                new MetricsParser.Dimension("clusterid", "CLUSTER-2"));
+
+        System.out.println(dimensionsHashCode(dimensions1));
+        System.out.println(dimensionsHashCode(dimensions2));
```

Comment:
I added test and also removed the 2 System.out.println lines assuming they were debug.